spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/ChatClientObservationContext.java

```diff
@@ -0,0 +1,60 @@
+/*
+* Copyright 2024 - 2024 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.chat.client.observation;
+
+import org.springframework.ai.chat.client.DefaultChatClient.DefaultChatClientRequestSpec;
+import org.springframework.ai.chat.model.ChatResponse;
+import org.springframework.ai.model.observation.ModelObservationContext;
+import org.springframework.ai.observation.AiOperationMetadata;
+
+/**
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+
+public class ChatClientObservationContext extends ModelObservationContext<DefaultChatClientRequestSpec, ChatResponse> {
```

Comment:
I'm not sure about defining this context as a subtype of `ModelObservationContext`, which is more focused on the "request/response" single interactions with a model service. It could be its own type without inheritance. What do you think?

Reply:
We might just need the `DefaultChatClientRequestSpec` object here, and perhaps additional metadata like `format` and `isStream`

Reply:
Agree. I was tempted because it thought we can reuse some code related to the ChatResponse. But this is not relevant.